### PR TITLE
Fix: Send TCP Forward Request reply

### LINKED ISSUE #118: ssh.client.ListenTCP hangs
Awesome app, thanks for the work.

I'm trying to write a golang client that uses this relay to establish a remote port forward, and when I try to start a port listener from a tcpclient with client.Listen or client.ListenTCP, the client blocks. If I do it against an openssh server, it works fine. Using `ssh.exe -R`  against the SISH server also works fine.

I tried connecting to the port to see if it continues it but it stays blocked, so I think something is wrong here (maybe a confirmation message not coming back correctly)?

## Steps to reproduce
```go
	cfg := &ssh.ClientConfig{
		User: "test",
		Auth: []ssh.AuthMethod{
			ssh.Password("test"),
		},
		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
	}
	client, err := ssh.Dial("tcp", "mytestserver:2222", cfg)

	listenAddress := &net.TCPAddr{
		IP:   net.ParseIP("127.0.0.1"),
		Port: 40022,
	}
	l, err := client.ListenTCP(listenAddress) //It hangs here but proceeds fine against a openssh server. No message on the server
```

### PR DESCRIPTION
**Pull request by pcj**
Continues the work of @JustinGrote.  Justin, feel free to cherry-pick this onto your branch if you like!  I also would like to use a golang ssh client, so it looks like an important change.

Closes #118

### CODE REVIEW

File: sshmuxer/requests.go [hunk 1]
```diff
@@ -171,6 +177,15 @@ func handleRemoteForward(newRequest *ssh.Request, sshConn *utils.SSHConnection,
 			return
 		}
 
+		// reply that the session is successfully set up with the port that was
+		// assigned (which may be different than what was requested)
+		channelForwardReplyPayload := ssh.Marshal(channelForwardReply{bindPort})
```
Comment (antoniomika) [inline sshmuxer/requests.go:182]: Just to confirm, does this work fine in situations with HTTP, TCP, and Alias forwards?

Comment (pcj) [reply sshmuxer/requests.go:182]: Thinking about how to write a proper test, either with the compiled binary or preferably a smaller subset of the components in this library...

Comment (antoniomika) [reply sshmuxer/requests.go:182]: Something like that would likely need to be tested with the compiled bin/with an SSH compliant client just to verify it does what is expected

**Comment by antoniomika** [comment]
Merging this in so I can apply some changes ontop! Thanks @pcj!

**Comment by pcj** [comment]
Apologies I never got to that test.  I did start on a framework to write integration tests for sish, but never finished and got sidetracked.  Sorry about that!

**Comment by antoniomika** [comment]
Appreciate that! I ended up taking some time to make sure this worked as we expected it (sometimes ssh forwards would fail as a different port was being forwarded than what was setup and I applied this to all of the tunnel types). Let me know if you try out the newest version how things are working for ya!

**Pull request merged**